lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -377,9 +377,17 @@ def get_or_create_cached_value():
                 messages = cached_func.message_call_stack._most_recent_messages
                 try:
                     cache.write_result(value_key, return_value, messages)
-                except TypeError:
-                    if type_util.is_type(
-                        return_value, "snowflake.snowpark.dataframe.DataFrame"
+                except (TypeError, RuntimeError):
+                    if (
+                        type_util.is_type(
+                            return_value, "snowflake.snowpark.table.Table"
+                        )
+                        or type_util.is_type(
+                            return_value, "snowflake.snowpark.dataframe.DataFrame"
+                        )
+                        or type_util.is_type(
+                            return_value, "pyspark.sql.dataframe.DataFrame"
+                        )
                     ):
 
                         class UnevaluatedDataFrameError(StreamlitAPIException):
```

Comment:
I know this isn't part of the PR, but can we move this class declaration outside the function - to the top of the file, or wherever?

Reply:
Sure 👍 It's moved now into `cache_errors.py` file 👍